app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -88,39 +95,100 @@ public SchedulePlan getSchedulePlan(StudyIdentifier studyIdentifier, String guid
         if (plans.isEmpty()) {
             throw new EntityNotFoundException(SchedulePlan.class);
         }
-        return plans.get(0);
+        
+        plan = plans.get(0);
+        forEachCriteria(plan, scheduleCriteria -> loadCriteria(scheduleCriteria));
+        return plan;
     }
 
     @Override
     public SchedulePlan createSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
-        checkNotNull(studyIdentifier, "StudyIdentifier is null");
-        checkNotNull(plan, "Schedule plan is null");
+        checkNotNull(studyIdentifier);
+        checkNotNull(plan);
         
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setGuid(BridgeUtils.generateGuid());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        
+        forEachCriteria(plan, scheduleCriteria -> persistCriteria(scheduleCriteria));
         mapper.save(plan);
         return plan;
     }
```

Comment:
It's probably worth thinking about how we store this in DDB. I feel like it would be cleanest if the Schedule Plan in DDB stored criteria as a list of criteria IDs (or maybe even a count of criteria, since the IDs all follow the same pattern). When we read it from DDB, we query the Criteria table for each criteria ID and populate it into the Schedule Plan.

When we create or save CriteriaSchedulePlanStrategies, the JSON includes the criteria in place, and we strip out the Criteria, save them individually to the Criteria table, and write the list of criteria IDs (or counts) into the Strategy.

This way, we can avoid saving the same thing to DDB twice.